### PR TITLE
ESP32S2 camera board draft

### PR DESCRIPTION
**Pull request by ladyada**

### CODE REVIEW

**Comment by jepler** [comment]
looks like this'll need the SPI fix we discussed for the feather esp32s2 tft, I think you know it, but just a reminder :)

File: ports/espressif/boards/adafruit_qtpy_esp32s3_nopsram/pins.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    { MP_ROM_QSTR(MP_QSTR_D34), MP_ROM_PTR(&pin_GPIO34) },
+
+    { MP_ROM_QSTR(MP_QSTR_SCK), MP_ROM_PTR(&pin_GPIO35) },
+    { MP_ROM_QSTR(MP_QSTR_D35), MP_ROM_PTR(&pin_GPIO35) },
+
+    { MP_ROM_QSTR(MP_QSTR_MISO), MP_ROM_PTR(&pin_GPIO36) },
+    { MP_ROM_QSTR(MP_QSTR_D36), MP_ROM_PTR(&pin_GPIO36) },
+
+    { MP_ROM_QSTR(MP_QSTR_NEOPIXEL_POWER), MP_ROM_PTR(&pin_GPIO37) },
+    { MP_ROM_QSTR(MP_QSTR_NEOPIXEL), MP_ROM_PTR(&pin_GPIO38) },
```
Comment (tannewt) [inline ports/espressif/boards/adafruit_qtpy_esp32s3_nopsram/pins.c:47]: Is this QT Py ESP32-S3 rev B? The schematic as I see it has NEOPIXEL_POWER on IO38 and NEOPIX in MTCK.

File: ports/espressif/boards/adafruit_qtpy_esp32s3_nopsram/pins.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    { MP_ROM_QSTR(MP_QSTR_A3), MP_ROM_PTR(&pin_GPIO8) },
+    { MP_ROM_QSTR(MP_QSTR_D8), MP_ROM_PTR(&pin_GPIO8) },
+
+    { MP_ROM_QSTR(MP_QSTR_SDA), MP_ROM_PTR(&pin_GPIO7) },
+    { MP_ROM_QSTR(MP_QSTR_A4), MP_ROM_PTR(&pin_GPIO7) },
+    { MP_ROM_QSTR(MP_QSTR_D7), MP_ROM_PTR(&pin_GPIO7) },
+
+    { MP_ROM_QSTR(MP_QSTR_SCL), MP_ROM_PTR(&pin_GPIO6) },
+    { MP_ROM_QSTR(MP_QSTR_A5), MP_ROM_PTR(&pin_GPIO5) },
```
Comment (tannewt) [inline ports/espressif/boards/adafruit_qtpy_esp32s3_nopsram/pins.c:26]:
```suggestion
    { MP_ROM_QSTR(MP_QSTR_A5), MP_ROM_PTR(&pin_GPIO6) },
```

File: ports/espressif/boards/adafruit_qtpy_esp32s3_nopsram/pins.c [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    { MP_ROM_QSTR(MP_QSTR_D16), MP_ROM_PTR(&pin_GPIO16) },
+
+    { MP_ROM_QSTR(MP_QSTR_MOSI), MP_ROM_PTR(&pin_GPIO34) },
+    { MP_ROM_QSTR(MP_QSTR_D34), MP_ROM_PTR(&pin_GPIO34) },
+
+    { MP_ROM_QSTR(MP_QSTR_SCK), MP_ROM_PTR(&pin_GPIO35) },
+    { MP_ROM_QSTR(MP_QSTR_D35), MP_ROM_PTR(&pin_GPIO35) },
+
+    { MP_ROM_QSTR(MP_QSTR_MISO), MP_ROM_PTR(&pin_GPIO36) },
+    { MP_ROM_QSTR(MP_QSTR_D36), MP_ROM_PTR(&pin_GPIO36) },
```
Comment (tannewt) [inline ports/espressif/boards/adafruit_qtpy_esp32s3_nopsram/pins.c:44]: I think these are shift in rev B.

File: ports/espressif/boards/adafruit_esp32s2_camera/pins.c [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        MP_ROM_PTR(&pin_GPIO12),
+        MP_ROM_PTR(&pin_GPIO10),
+        MP_ROM_PTR(&pin_GPIO9),
+        MP_ROM_PTR(&pin_GPIO7),
+    }
+};
+
+
+STATIC const mp_rom_map_elem_t board_module_globals_table[] = {
+    CIRCUITPYTHON_BOARD_DICT_STANDARD_ITEMS
```
Comment (tannewt) [inline ports/espressif/boards/adafruit_esp32s2_camera/pins.c:22]: Is this board in the board repo? I didn't see it.

**Review by tannewt — CHANGES_REQUESTED** [review]
I think the S3 qtpy pins are for an earlier rev. I wasn't able to check the camera board pins.

File: ports/espressif/boards/adafruit_qtpy_esp32s3_nopsram/pins.c (hunk 1 above)
Comment (ladyada) [reply ports/espressif/boards/adafruit_qtpy_esp32s3_nopsram/pins.c:47]: no this is rev A - which im the only owner of :) when the rev B shows up ill change it.

File: ports/espressif/boards/adafruit_qtpy_esp32s3_nopsram/pins.c (hunk 3 above)
Comment (ladyada) [reply ports/espressif/boards/adafruit_qtpy_esp32s3_nopsram/pins.c:44]: yep i got these wrong in rev A, whee the QFN was shifted one pin over. will update when rev B arrives

File: ports/espressif/boards/adafruit_esp32s2_camera/pins.c (hunk 4 above)
Comment (ladyada) [reply ports/espressif/boards/adafruit_esp32s2_camera/pins.c:22]: Development\Dev Boards\BrainCraft\Pylaroid

File: ports/espressif/boards/adafruit_qtpy_esp32s3_nopsram/pins.c (hunk 2 above)
Comment (ladyada) [reply ports/espressif/boards/adafruit_qtpy_esp32s3_nopsram/pins.c:26]: woops! fixed in latest commit

**Review by tannewt — APPROVED** [review]
Ah, ok. This looks good to me!

**Pull request merged**